source/connection_manager.c

```diff
@@ -812,6 +812,13 @@ struct aws_http_connection_manager *aws_http_connection_manager_new(
         return NULL;
     }
 
+    if (options->tls_connection_options && options->prior_knowledge_http2) {
+        AWS_LOGF_ERROR(
+            AWS_LS_HTTP_CONNECTION_MANAGER, "Invalid options - prior knowledge cannot be set when TLS is used");
```

Comment:
Is it an error if neither `tls_connection_options` or `prior_knowledge_http2` is set? because we don't support Upgrade from HTTP/1?

Reply:
related:
should we support prior-knowledge with TLS? Like, if TLS is used but not ALPN? Is that even a thing? We should only bother trying to support it if we have an actual internal use-case